lib/proxyquire.js

```diff
@@ -161,18 +190,29 @@ Proxyquire.prototype._require = function(module, stubs, path) {
 
   // Only ignore the cache if we have global stubs
   if (this._containsRuntimeGlobal) {
-    return this._withoutCache(module, stubs, path, Module._load.bind(Module, path, module));
+    return this._withoutCache(module, resolvedStubs, path, Module._load.bind(Module, path, module));
   } else {
     return Module._load(path, module);
   }
 };
 
+// `module` here is the module that's invoking proxyquire.
+// `path` is the path of the module we're going to require.
+//
 Proxyquire.prototype._withoutCache = function(module, stubs, path, func) {
   // Temporarily disable the cache - either per-module or globally if we have global stubs
   var restoreCache = this._disableCache(module, path);
+  var resolvedPath = Module._resolveFilename(path, module);
+  var self = this;
```

Comment:
Don't use `self = this` except when necessary. `[].reduce` and its derivatives (e.g. `map`) take a `thisArg` as their last argument.

Reply:
Really?  The [MDN page doesn't mention this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce).  In ES6 I would just use an arrow function.  I suppose I could just use 'bind'.

Reply:
Hmm I guess it's _only_ reduce that's missing it. It has an extra arg (the initial value) but I always consider it the "base" list function since it's the only one you can use to make all the others without loops or closures. Feel free to use `self = this` for `[].reduce` or `bind`. No perf concern here so either is ok. 

I wouldn't mind adopting some light ES6 syntax (node 4 compatible) sometime in the future but let's not change that now.